Daily Rewards

dieoska opened the pull request:
Everything is ready. It was a little hard to deal with the pull, but everything seemed to be fine.

dieoska committed:
Add daily reward date to account data

dieoska committed:
Add daily reward chat command

dieoska committed:
Add configurable daily reward money

dieoska committed:
Add daily reward status to account info

dieoska committed:
Fix daily reward declaration duplicates

dieoska committed:
Fix daily reward duplicates

dieoska commented:
@fokkonaut )

fokkonaut commented on src/game/server/ddracechat.cpp:
```diff
@@ -1841,6 +1862,56 @@ void CGameContext::ConPin(IConsole::IResult* pResult, void* pUserData)
+	pSelf->WriteAccountStats(ID);
+
+	char aBuf[128];
+	str_format(aBuf, sizeof(aBuf), pPlayer->Localize("Вы получили ежедневную награду: %d монет"), Reward);
```
Translation

fokkonaut commented on src/game/server/gamecontext.cpp:
```diff
@@ -6093,7 +6095,8 @@ void CGameContext::ReadAccountStats(int ID, const char *pName)
 	{
-		getline(AccFile, data);
+		if (!getline(AccFile, data))
+			data.clear();
```
Did you run into any issues, or why was this required for this feature?

fokkonaut commented on src/game/server/votingmenu.cpp:
```diff
@@ -646,6 +646,26 @@ void CVotingMenu::DoPageAccount(int ClientID, int *pNumOptions)
+			str_format(aBuf, sizeof(aBuf), "Ежедневная награда: %s", pDailyStatus);
+		}
+		DoLineText(Page, pNumOptions, aBuf);
+
```
Same as in ConAccount

dieoska replied:
Thanks for the feedback, makes sense.

I'll address the localization, adjust the date logic for daily rewards, remove the duplicated reset, and clean up the extra changes. Will push an updated version soon.

dieoska committed:
Fix daily reward localization and logic

dieoska committed:
Merge pull request #25 from NullSeeker/codex/fix-review-feedback-on-pr-#395

Fix daily reward localization and logic

dieoska committed:
Make GetDate const-correct

dieoska committed:
Merge pull request #26 from NullSeeker/codex/fix-cgamecontext-const-qualifier-errors

Make `CGameContext::GetDate` const-correct

dieoska commented:
@fokkonaut Done. All that's left is to translate the text in the txt files themselves; I didn't really understand how they worked.

dieoska committed:
Add daily reward localization strings

dieoska committed:
update launguages

Add daily reward localization entries to language files

dieoska committed:
Localize daily command help

dieoska committed:
test

Localize chat command help and daily reward text

fokkonaut commented on src/game/server/ddracechat.h:
```diff
@@ -62,11 +62,13 @@ CHAT_COMMAND("spawnweapons", "", CFGFLAG_CHAT, ConSpawnWeaponsInfo, this, "Shows
 CHAT_COMMAND("pin", "?s[pin]", CFGFLAG_CHAT, ConPin, this, "Sets a new security pin for your account", AUTHED_NO)
 
 CHAT_COMMAND("pay", "i[amount] r[name]", CFGFLAG_CHAT, ConPayMoney, this, "Pays i money to player r", AUTHED_NO)
+CHAT_COMMAND("dailyreward", "", CFGFLAG_CHAT, ConDailyReward, this, "Daily reward", AUTHED_NO)
```
why 2 commands?

dieoska replied:
Initially, I wanted to add other features to /dailyreward, but later changed my mind and forgot to remove them after testing.

fokkonaut commented on src/game/server/gamecontext.cpp:
```diff
@@ -969,26 +969,34 @@ void CGameContext::SendGameMsg(int GameMsgID, int ParaI1, int ParaI2, int ParaI3
+	{
+		CPlayer *pPlayer = m_apPlayers[ClientID];
+		if (pPlayer)
+			pHelpText = pPlayer->Localize(pCommand->m_aHelpText);
```
As said, i dont think we should translate help text in this PR. It's more technical and can cause issues when non-english

fokkonaut commented on src/game/server/gamecontext.cpp:
```diff
@@ -6613,6 +6627,39 @@ const char *CGameContext::GetDate(time_t Time, bool ShowTime)
+{
+	struct tm FirstDate = *localtime(&First);
+	struct tm SecondDate = *localtime(&Second);
+	return FirstDate.tm_year == SecondDate.tm_year && FirstDate.tm_yday == SecondDate.tm_yday;
```
Month missing?

dieoska committed:
Adjust daily reward command and date checks

dieoska committed:
remove double command and fix date check

Fix daily reward: remove duplicate command, correct date check, and reorder reset

dieoska committed:
Remove daily reward translations

dieoska committed:
перевод убран

Remove daily reward translations

dieoska commented:
@fokkonaut You can check, 1 more request, if it's not difficult, you can describe the edit in more detail. I'm not very good with English, and the translator doesn't translate well.

fokkonaut commented on datasrc/languages/bulgarian.txt:
```diff
@@ -2093,3 +2093,6 @@ Your stake got returned: +%lld money.
 == 
+
+available
+== налична
```
There is still one translation in each language file which shouldn't be there :D

fokkonaut replied:
<img width="227" height="85" alt="grafik" src="https://github.com/user-attachments/assets/ac026c5d-7440-48c5-b32e-bd5908514532" />

Check *each* file

fokkonaut commented on src/game/server/ddracechat.cpp:
```diff
@@ -92,7 +93,7 @@ void CGameContext::ConHelp(IConsole::IResult *pResult, void *pUserData)
 
 			if (pCmdInfo->m_pHelp)
-				pSelf->Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "help", pCmdInfo->m_pHelp);
+				pSelf->Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "help", pPlayer->Localize(pCmdInfo->m_pHelp));
```
Help texts are currently not translated

dieoska committed:
delete 'налична'

dieoska commented:
@fokkonaut this all change ?

dieoska commented on datasrc/languages/bulgarian.txt:
```diff
@@ -2093,3 +2093,6 @@ Your stake got returned: +%lld money.
 == 
+
+available
+== налична
```
okay

dieoska committed:
remove translate 'help'

dieoska committed:
delete translate from all launguage files

dieoska commented:
@fokkonaut okay all ready

dieoska committed:
fix warning

dieoska commented:
@fokkonaut ?

fokkonaut commented:
I still see unnecessary code changes:
- newlines in language files
- changes in `CGameContext::SendChatCommand`
- changes in whois

Other than that lgtm

dieoska commented:
@fokkonaut Can you please explain in more detail, I didn't really understand what needs to be fixed?

ChillerDragon commented on datasrc/languages/arabic.txt:
```diff
@@ -2093,3 +2093,4 @@ Your stake got returned: +%lld money.
 
 You got +%lld money from others leaving the game.
 == 
+
```
@NullSeeker this newline is what fokko is complaining about

dieoska committed:
delete line

dieoska committed:
f

dieoska commented:
@ChillerDragon okay, this all ?

dieoska committed:
clear 'CGameContext::SendChatCommand', clear whois

dieoska commented:
@fokkonaut done

dieoska commented:
@fokkonaut ?

dieoska commented:
Happy New Year )
